lib/active_admin/middleware/error_handler.rb

```diff
@@ -0,0 +1,29 @@
+module ActiveAdmin
+  module Middleware
+    class ErrorHandler
+
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env)
+        @app.call(env)
+      rescue Exception => e
+        if ActiveAdmin.application.default_namespace.to_s == env['PATH_INFO'].split('/').second
```

Comment:
According to Rails/Engine docs, a rails app is supposed to create a separate middleware stack for the app and the engine. But this is not happening here. This is the reason, why I had to do it this way.

Reference: http://api.rubyonrails.org/classes/Rails/Engine.html#class-Rails::Engine-label-Configuration